dotnet/Cucumber.Messages/CucumberMessagEnumConverter.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Reflection;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace Cucumber.Messages
+{
+    public class CucumberMessagEnumConverter<T> : JsonConverter<T> where T : struct, Enum
```

Comment:
Versioning is easier if a library only exposes the essentials. Consider making this and any other classes not explicitly intended to be part of the API package private (or the .Net) equivalent.

Reply:
Agreed. I'll make this class internal to the Cucumber.Messages assembly.